conda/core/index.py

```diff
@@ -56,14 +56,7 @@ def supplement_index_with_prefix(index, prefix, channels):
     assert prefix
     maxp = len(channels) + 1
     for dist, info in iteritems(linked_data(prefix)):
-        if dist in index:
-            # The downloaded repodata takes priority, so we do not overwrite.
-            # We do, however, copy the link information so that the solver
-            # knows this package is installed.
-            old_record = index[dist]
-            link = info.get('link') or EMPTY_LINK
-            index[dist] = IndexRecord.from_objects(old_record, link=link)
-        else:
```

Comment:
we don't need this anymore if we get rid of the `installed` property on resolve, and then use `linked_data` appropriately